File: project/app-domain/src/main/java/com/numberbox/appdomain/annotation/Priority.java
package com.numberbox.appdomain.annotation;

import java.lang.annotation.*;

/**
 * Def. 같은 타입의 객체 중 의존 주입에 우선권을 갖음
 */
@Target({ElementType.TYPE, ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface Priority {
}

File: project/app-domain/src/main/kotlin/com/kamcci/numberbox/app/domain/system_construction/Priority.kt
package com.kamcci.numberbox.app.domain.system_construction

import java.lang.annotation.*
import java.lang.annotation.Retention
import java.lang.annotation.Target

/**
 * Def. 상위 타입의 하위 구현체 중 최우선 순위를 주기 위해 사용할 어노테이션
 * Desc. 스프링의 @Primary 역할
 */
@Target(ElementType.TYPE, ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
@Documented
annotation class Priority